content/file/errors.go

```diff
@@ -23,6 +23,7 @@ var (
 	ErrPathTraversalDisallowed = errors.New("path traversal disallowed")
 	ErrOverwriteDisallowed     = errors.New("overwrite disallowed")
 	ErrStoreClosed             = errors.New("store already closed")
+	ErrPreservePermissions     = errors.New("failed to restore permission during unpacking")
```

Comment:
Since this error is file store specific, it should be put the `file` package.
BTW, should it be "restore permission**s**" in the error message?

Reply:
this _is_ the file package.

>BTW, should it be "restore permissions" in the error message?

yep, typo

Reply:
btw, not sure how to increase the code coverage for this.
To do that, i think i would need to test the chmod error case. It isn't obvious to me how to trigger such failure, other than concurrently trying to delete a file before its chmod'ed or something (i.e. a race, but that would be flaky test)

Reply:
> this is the file package. (?)

Oh sorry I was blind😂
Yeah it would be very tricky to test the chmod error case... I think we can bypass it. /cc @shizhMSFT

Reply:
How do I bypass it?
do I ignore the entire file?

Reply:
We will ignore the patch CI failure and force merge...